typing/typedtree.mli

```diff
@@ -104,10 +106,19 @@ and pattern_desc =
             [row_desc] = [Some _] when translating [Ppat_type _],
                          [None] otherwise.
          *)
-  | Tpat_lazy of pattern
-        (** lazy P *)
-  | Tpat_exception of pattern
+  | Tpat_lazy of computation_pattern
+        (** lazy CP *)
+
+and computation_pattern =
+  computation_pattern_desc pattern_
+
+and computation_pattern_desc =
+  | Tcomp_pat_exception of pattern
         (** exception P *)
+  | Tcomp_pat_return of pattern
```

Comment:
Looks like monadic code... Wouldn't `Tcomp_pat_value` be what you meant?

Reply:
I have seen effect-handler languages where the syntax looks like

```
match e with
| return p -> ...
| Set v k -> ...
| ...
```

and it is this `return` usage that I am referring to (we are matching on a value that was returned by the expression, rather than an effectful operation that was raised). `value` would also be okay, I guess?